polkadot/node/core/pvf/src/security.rs

```diff
@@ -70,6 +65,60 @@ pub async fn check_security_status(config: &Config) -> Result<SecurityStatus, St
 	Ok(security_status)
 }
 
+/// Contains the full security status including error states.
+struct FullSecurityStatus {
+	partial: SecurityStatus,
```

Comment:
full/partial is rather confusing naming to me.

I think it would be better if `SecurityStatus` would have the `SecureModeResult` as fields instead of the booleans. I assume errors are strictly correlated to the security features

Reply:
Yeah, I had thought about that. What I didn't like about it is that instead of just looping over errors (which we do in a couple places) you have to check each struct field manually. That's less nice and also more error-prone, because I don't know how to do that exhaustively.

I don't know if I did it an ideal way, but `FullSecurityStatus` is designed to be easy and safe to extend, with a hopefully-nice API. And we are already thinking about [extending the security features](https://github.com/paritytech/polkadot-sdk/pull/2477#discussion_r1406110772).

Another point, is that the exact errors are not needed and unused elsewhere, so storing them in `SecurityStatus` would just make it harder to use.